crates/sui-core/src/consensus_adapter.rs

```diff
@@ -184,6 +192,33 @@ impl ConsensusAdapterMetrics {
     }
 }
 
+pub enum BlockStatus {
+    Sequenced,
+    GarbageCollected,
+}
+
+pub enum SubmitResponse {
```

Comment:
I'm not sure why the `NoWait` variant is needed.

Reply:
This is currently used for unit testing as it makes it easier to just return the result directly instead of crafting oneshot responses etc. But I am thinking now that maybe could refactor this a little bit. Let me look again